### PR TITLE
fix opensearch mapper_parsing_exception

### LINKED ISSUE #2335: Karmada search sync data error By using opensearch
**What happened**:
Karmada search sync data error By using opensearch
**What you expected to happen**:
Karmada search sync data successfully By using opensearch
**How to reproduce it (as minimally and precisely as possible)**:
1、using opensearch to registry
image
2、member clusters has developments with label `app.kubernetes.io/managed-by` and so on
3、search log
image

**Anything else we need to know?**:

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

### PR DESCRIPTION
**Pull request by liys87x**
karmada-search error log: Could not dynamically add mapping for field [app.name]. Existing mapping for [metadata.labels.app] must be of type object but found [text].

Co-authored-by: liys87x <liyasong1987x@gmail.com>
Signed-off-by: huntsman_ly <huntsman_ly@sina.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #2335 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by liys87x** [comment]
Example of mapper_parsing_exception error generated under dynamic mapping

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: nginx
  name: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - image: nginx:alpine
        name: nginx
```

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.name: nginx1
  name: nginx1
spec:
  replicas: 3
  selector:
    matchLabels:
      app.name: nginx1
  template:
    metadata:
      labels:
        app.name: nginx1
    spec:
      containers:
      - image: nginx:alpine
        name: nginx1
```

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/cc @wuyingjun-lucky Can you help take a look and have a test?

**Comment by wuyingjun-lucky** [comment]
> /cc @wuyingjun-lucky Can you help take a look and have a test?

OK, I will finish it this weekend.
/assign

**Comment by wuyingjun-lucky** [comment]
I have tested it works well on my side @XiShanYongYe-Chang.
image
I am not sure whether it will have an effect on query actions like labelselector if we change `flattened` to `object` and set it to false although it seems we have not implements query actions on opensearch backend.   
Leave it to you @liys87x @RainbowMango  @XiShanYongYe-Chang.

File: pkg/search/backendstore/opensearch.go [hunk 1]
```diff
@@ -232,27 +236,31 @@ func (os *OpenSearch) indexName(us *unstructured.Unstructured) (string, error) {
 	os.l.Lock()
 	defer os.l.Unlock()
 
-	if _, ok := os.indices[name]; !ok {
```
Comment (liys87x) [inline pkg/search/backendstore/opensearch.go:235]: The mapping settings before the modification did not take effect, and dynamic mapping was used.

**Comment by liys87x** [comment]
opensearch does not support `flattened`
https://github.com/opensearch-project/OpenSearch/issues/1018

**Comment by wuyingjun-lucky** [comment]
> opensearch does not support `flattened` [opensearch-project/OpenSearch#1018](https://github.com/opensearch-project/OpenSearch/issues/1018)

OK looks good to me
cc @XiShanYongYe-Chang

**Comment by XiShanYongYe-Chang** [comment]
@wuyingjun-lucky Thanks a lot!

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Let's move it on.
/lgtm
/approve

**Pull request merged**